src/CSAccounting.sol

```diff
@@ -658,18 +743,27 @@ contract CSAccounting is
         revert NodeOperatorDoesNotExist();
     }
 
-    function _onlyNodeOperatorManagerOrRewardAddresses(
-        NodeOperatorManagementProperties memory no
-    ) internal view {
+    function _checkAndGetEligibleNodeOperatorProperties(
+        uint256 nodeOperatorId
+    ) internal view returns (NodeOperatorManagementProperties memory no) {
+        no = MODULE.getNodeOperatorManagementProperties(nodeOperatorId);
         if (no.managerAddress == address(0)) {
             revert NodeOperatorDoesNotExist();
         }
 
-        if (no.managerAddress == msg.sender || no.rewardAddress == msg.sender) {
-            return;
+        if (no.managerAddress != msg.sender && no.rewardAddress != msg.sender) {
+            revert SenderIsNotEligible();
         }
+    }
 
-        revert SenderIsNotEligible();
+    /// @dev Bond reserve can be removed if a sufficient time has passed or if the Node Operator has no active or depositable keys
+    function _canRemoveBondReserve(
```

Comment:
it think it should be either a public function or an internal function that reverts
see no reason to have it internal boolean